lib/Core/Executor.cpp

```diff
@@ -3216,6 +3320,14 @@ void Executor::executeMemoryOperation(ExecutionState &state,
       value = state.constraints.simplifyExpr(value);
   }
 
+  if ((OptimizeArray == VALUE) && !isa<ConstantExpr>(address)) {
```

Comment:
Shouldn't the first part be `(OptimizeArray == VALUE || OptimizeArray == ALL)` here?